addons/sale_stock/sale_stock.py

```diff
@@ -355,6 +355,16 @@ def product_id_change_with_wh(self, cr, uid, ids, pricelist, product, qty=0,
         res.update({'warning': warning})
         return res
 
+    def button_cancel(self, cr, uid, ids, context=None):
+        lines = self.browse(cr, uid, ids, context=context)
+        procurement_obj = self.pool['procurement.order']
+        move_obj = self.pool['stock.move']
+        for procurement in procurement_obj.browse(cr, uid, sum([l.procurement_ids.ids for l in lines], []), context=context):
```

Comment:
browse seems unnecessary.  In old api, l.procurement_ids gives a list with browse records.